### PR TITLE
Replace pickle by json

### PR DESCRIPTION
**Pull request by maipbui**
#### What I did
`pickle` can lead to lead to code execution vulnerabilities. Recommend to serializing the relevant data as JSON.
#### How I did it
Replace `pickle` by `json`
#### How to verify it
Pass UT
Manual test
Execute these commands in DUT
```
  	1. dropstat
		a. show dropcounters capabilities
		b. show dropcounters configuration
		c. show dropcounters counts
		d. sonic-clear dropcounters
	2. flow_counter_stat
		a. sonic-clear flowcnt-trap
		b. show flowcnt-trap stats
	3. intfstat
		a. sonic-clear rifcounters
		b. show interfaces counters rif
		c. show interfaces counters rif -p 2
		d. show interfaces counters rif -p 2
	4. pfcstat
		a. show pfc counters
		b. sonic-clear pfccounters
	5. pg-drop
		a. show priority-group drop counters 
		b. sonic-clear priority-group drop counters
	6. portstat
		a. show interfaces counters
		b. show interfaces counters errors
		c. show interfaces counters errors -p 2
		d. show interfaces counters rates
		e. show interfaces counters rates -p 2
		f. show interfaces counters detailed
		g. show interfaces counters detailed -p 2
		h. sonic-clear counters
	7. queuestat
		a. sonic-clear queuecounters
		b. show queue counters
		c. show queue counters -p 2
	8. tunnelstat
		a. sonic-clear tunnelcounters
		b. show vxlan counters
		c. show vxlan counters -p 2
```

### CODE REVIEW

File: scripts/dropstat [hunk 1]
```diff
@@ -117,10 +117,10 @@ class DropStat(object):
         """
 
         try:
-            pickle.dump(self.get_counts_table(self.gather_counters(std_port_rx_counters + std_port_tx_counters, DEBUG_COUNTER_PORT_STAT_MAP), COUNTERS_PORT_NAME_MAP),
-                        open(self.port_drop_stats_file, 'wb+'))
-            pickle.dump(self.get_counts(self.gather_counters([], DEBUG_COUNTER_SWITCH_STAT_MAP), self.get_switch_id()),
-                        open(self.switch_drop_stats_file, 'wb+'))
+            json.dump(self.get_counts_table(self.gather_counters(std_port_rx_counters + std_port_tx_counters, DEBUG_COUNTER_PORT_STAT_MAP), COUNTERS_PORT_NAME_MAP),
```
Comment (qiluo-msft) [inline scripts/dropstat:120]: >json [](http://example.com/codeflow?start=12&length=4)

The behavior is different, and we need to very careful. Suggest further separater pickle related code to a smaller PR and ask for more reviewers.

**Review by Junchao-Mellanox — APPROVED** [review]

**Comment by qiluo-msft** [comment]
@vaibhavhd Are the pickle dump files used after warm upgrade or fast upgrade?

**Review by qiluo-msft — APPROVED** [review]

**Comment by maipbui** [comment]
@vaibhavhd could you help review?

**Comment by vaibhavhd** [comment]
> @vaibhavhd Are the pickle dump files used after warm upgrade or fast upgrade?

I don't think the dump files modified here are used by warm/fast upgrade.

**Review by vaibhavhd — APPROVED** [review]

**Review by dgsudharsan — APPROVED** [review]

**Comment by prgeor** [comment]
@maipbui any reference you can provide for pickle security vulnerabilities?

**Comment by maipbui** [comment]
> @maipbui any reference you can provide for pickle security vulnerabilities?

Here are some docs:
https://semgrep.dev/r?q=python.lang.security.deserialization.pickle.avoid-pickle
https://docs.python.org/3/library/pickle.html
https://github.com/dlint-py/dlint/blob/master/docs/linters/DUO103.md
https://learning.oreilly.com/library/view/mastering-python-for/9781839217166/B15676_11_ePub_AM.xhtml#_idParaDest-252 section pickle module security

**Pull request merged**